src/groupByKey.ts

```diff
@@ -1,7 +1,26 @@
+import { error } from 'console';
+
 type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> | null {
+  const result: GroupsMap<T> = {};
+
+  for (const item of items) {
+    const name = (item[key] as string) || null;
```

Comment:
The expression `(item[key] as string) || null` assumes that the key will always be a string. This might not be the case for all types of `T`. Consider checking the type of `item[key]` before casting it to a string.